src/components/submission/diff-item.vue

```diff
@@ -0,0 +1,226 @@
+      </template>
+    </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'SubmissionDiffItem',
+  props: {
+    entry: {
+      type: Object,
+      required: true
+    },
+    parentPath: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+    fields: {
+      type: Object,
+      required: false
+    }
+  },
+  computed: {
+    isAtomicChange() {
+      // Is this a diff of a single field or whole subtree
+      if (this.entry.old === null)
+        if (typeof (this.entry.new) === 'object')
+          return false;
+      if (this.entry.new === null)
+        if (typeof (this.entry.old) === 'object')
+          return false;
+      return true;
+    },
+    visiblePath() {
+      if (!this.isAtomicChange)
+        return this.entry.path;
+      return this.entry.path.slice(0, this.entry.path.length - 1);
+    },
+    fieldName() {
+      return this.entry.path[this.entry.path.length - 1];
+    },
+    typeOfChange() {
+      if (this.entry.old === null)
+        return 'added';
+      if (this.entry.new === null)
+        return 'deleted';
+      return 'changed';
+    },
+    nestedDiffs() {
+      return this.flattenDiff(this.entry.new || this.entry.old, this.typeOfChange);
+    },
+    isBinary() {
+      // Compares path as array by converting it to /field1/field2 string and checks
+      // if it is for a binary field
+      const fullPath = this.parentPath.concat(this.entry.path);
+      const fullPathStr = fullPath.map((field) => (Array.isArray(field) ? field[0] : field)).join('/');
+      const basicPath = `/${fullPathStr}`;
+      if (this.fields[basicPath])
+        return true;
+      return false;
+    }
+  },
+  methods: {
+    flattenDiff(node, typeOfChange, prefix = []) {
+      // Flattens the inner diff (JSON of an entire added/removed tree structure)
+      // by recursively traversing the tree
+      let changes = [];
+      // eslint-disable-next-line guard-for-in
+      for (const k in node) {
+        const path = prefix.concat(k);
+        if (Array.isArray(node)) {
+          path.pop(); // pop off the index (e.g. 1 of toy #1)
+          path.pop(); // pop off the name (e.g. toy of toy #1)
+          path.push([prefix[prefix.length - 1], k]);
+        }
+        if (typeof (node[k]) === 'object') {
+          changes = changes.concat(this.flattenDiff(node[k], typeOfChange, path));
+        } else if (typeOfChange === 'added') {
+          changes.push({ path, old: null, new: node[k] });
+        } else {
+          changes.push({ path, old: node[k], new: null });
+        }
+      }
+      return changes;
+    },
+    binaryHref(value) {
+      return `/v1${this.$route.fullPath}/attachments/${value}`;
+    }
+  }
+};
+</script>
+
+<style lang="scss">
+@import '../../assets/scss/mixins';
+
+.submission-diff-item {
+  border-bottom: 1px solid #ccc;
+}
+
+.submission-diff-item:last-child {
+  border-bottom: 0;
+}
+
+.full-path, .field-name, .nested-change-type {
```

Comment:
You can nest class selectors within `.submission-diff-item` so that they're scoped to the component.